Server/htdocs/AppController/commands_RSM/api/v2/items/update.php

```diff
@@ -0,0 +1,106 @@
+require_once '../../../utilities/RSMverifyBody.php';
+setAuthorizationTokenOnGlobals();
+checkCorrectRequestMethod('PATCH');
+
+require_once '../../../utilities/RSdatabase.php';
+require_once '../../../utilities/RSMitemsManagement.php';
+require_once '../../api_headers.php';
+
+$requestBody = getRequestBody();
+verifyBodyContent($requestBody);
+
+// Definitions
+$clientID = getClientID();
+$RStoken = getRStoken();
+$RSuserID = getRSuserID();
+
+foreach ($requestBody as $item) {
+  $propertiesID = array();
+  //Iterate through every propertyID of the items to check if they pertain to the same item type
+  foreach ($item as $propertyID => $propertyValue) {
+    if ($propertyID != 'ID') {
+      $propertiesID[] = ParsePID($propertyID, $clientID);
+    }
+  }
+  $itemTypeID = getItemTypeIDFromProperties($propertiesID, $clientID);
+  foreach ($propertiesID as $propertyID) {
+    if (!(RShasTokenPermission($RStoken, $propertyID, 'WRITE') || isPropertyVisible($RSuserID, $propertyID, $clientID))) {
+      $RSallowDebug ? returnJsonMessage(400, 'Not Updated (At least 1 property has no WRITE permissions or is not visible)')  : returnJsonMessage(400, '');
+    }
+  }
+  $itemID = $item->ID;
+  if ($itemTypeID == 0) {
+    $RSallowDebug ? returnJsonMessage(400, 'Not Updated (Properties must pertain to the same item type)') : returnJsonMessage(400, '');
+  } elseif (!verifyItemExists($itemID, $itemTypeID, $clientID)) {
+    $RSallowDebug ? returnJsonMessage(400, 'Item doesn\'t exist') : returnJsonMessage(400, '');
+  }
+}
+
+foreach ($requestBody as $item) {
+  $itemTypeID = getItemTypeIDFromProperties($propertiesID, $clientID);
+  $itemID = $item->ID;
+  foreach ($item as $propertyID => $propertyValue) {
+    if ($propertyID != 'ID') {
+      $id = ParsePID($propertyID, $clientID);
+      $propertyType = getPropertyType($id, $clientID);
+      if (($propertyType == 'file') || ($propertyType == 'image')) {
+        $pieces = explode(":", $propertyValue);
+        if (count($pieces) == 1) {
+          $name = "";
+          $value = $pieces[0];
+        } else {
+          $name = $pieces[0];
+          $value = $pieces[1];
+        }
+
+        if ($value == "") {
+          deleteItemPropertyValue($itemTypeID, $itemID, $id, $clientID, $propertyType);
+        } else {
+          $result = setDataPropertyValueByID($id, $itemTypeID, $itemID, $clientID, $name, $value, $propertyType, $RSuserID);
+        }
+      } else {
+        if (!mb_check_encoding($propertyValue, 'UTF-8')) {
+          $RSallowDebug ? returnJsonMessage(400, 'Decoded parameter:' . $propertyValue . ' is not encoded in UTF-8') : returnJsonMessage(400, '');
+        }
+        $result = setPropertyValueByID($id, $itemTypeID, $itemID, $clientID, replaceUtf8Characters($propertyValue), $propertyType);
+      }
+      // Result = 0 is a successful response
+      if ($result != 0) {
+        $RSallowDebug ? returnJsonMessage(400, 'Not updated') : returnJsonMessage(400, '');
```

Comment:
If debug is enabled, add extra info, like the ID of the property that was not updated, the new value, etc...